conans/client/installer.py

```diff
@@ -36,76 +36,225 @@ def _init_package_info(deps_graph, paths, current_path):
+
+        finally:
+            self._conan_file._conanfile_directory = export_folder
+            # Now remove all files that were imported with imports()
+            for f in copied_files:
+                try:
+                    if f.startswith(self.build_folder):
+                        os.remove(f)
+                except Exception:
+                    self._out.warn("Unable to remove imported file from build: %s" % f)
+
+    @property
+    def build_folder(self):
+        if not self._build_folder:
+            new_id = build_id(self._conan_file)
+            new_ref = PackageReference(self._conan_ref, new_id) if new_id else self._package_reference
+            self._build_folder = self._client_cache.build(new_ref, self._conan_file.short_paths)
+        return self._build_folder
+
+
+def _raise_package_not_found_error(conan_file, conan_ref, out):
+    settings_text = ", ".join(conan_file.info.full_settings.dumps().splitlines())
+    options_text = ", ".join(conan_file.info.full_options.dumps().splitlines())
+
+    out.warn('''Can't find a '%s' package for the specified options and settings:
+- Settings: %s
+- Options: %s
+''' % (conan_ref, settings_text, options_text))
+
+    raise ConanException('''Missing prebuilt package for '%s'
+Try to build it from sources with "--build %s"
+Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
+''' % (conan_ref, conan_ref.name))
+
+
+def _handle_system_requirements(conan_file, package_reference, client_cache, out):
+    """ check first the system_reqs/system_requirements.txt existence, if not existing
+    check package/sha1/
+
+    Used after remote package retrieving and before package building
+    """
+    if "system_requirements" not in type(conan_file).__dict__:
+        return
+
+    system_reqs_path = client_cache.system_reqs(package_reference.conan)
+    system_reqs_package_path = client_cache.system_reqs_package(package_reference)
+    if os.path.exists(system_reqs_path) or os.path.exists(system_reqs_package_path):
+        return
+
+    ret = call_system_requirements(conan_file, out)
+
+    try:
+        ret = str(ret or "")
+    except:
+        out.warn("System requirements didn't return a string")
+        ret = ""
+    if getattr(conan_file, "global_system_requirements", None):
+        save(system_reqs_path, ret)
+    else:
+        save(system_reqs_package_path, ret)
+
+
+def call_system_requirements(conanfile, output):
+    try:
+        return conanfile.system_requirements()
+    except Exception as e:
+        output.error("while executing system_requirements(): %s" % str(e))
+        raise ConanException("Error in system requirements")
+
+
+def call_package_info(conanfile):
+    # Once the node is build, execute package info, so it has access to the
+    # package folder and artifacts
+    with conanfile_exception_formatter(str(conanfile), "package_info"):
+        conanfile.package_info()
 
 
 class ConanInstaller(object):
```

Comment:
Only managing graph and remote stuff, the building and packaging was extracted to the previous class.